components/eamxx/src/share/field/field_impl.hpp

```diff
@@ -372,9 +372,10 @@ update (const Field& x, const ST alpha, const ST beta)
 
   // Determine if there is a FillValue that requires extra treatment.
   ST fill_val = constants::DefaultFillValue<ST>().value;
-  const auto& xtra_data = get_header().get_extra_data();
+  const auto& xtra_data = x.get_header().get_extra_data();
+
   if (xtra_data.count("mask_value")) {
-    fill_val = ekat::any_cast<ST>(xtra_data.at("mask_value"));
+    fill_val = ekat::any_cast<float>(xtra_data.at("mask_value"));
```

Comment:
Since `update_impl` expects `fill_value` to be of same type of alpha/beta, we should at least check that the field data type is not `IntType`, otherwise when `update_impl` is called there will be a narrowing conversion.